src/dmifn.rs

```diff
@@ -1121,7 +1121,15 @@ pub fn dmi_memory_device_size(size: MemorySize) {
             println!("Error, extended Size does not exist.")
         }
         MemorySize::Kilobytes(size_kb) => println!("{} kB", size_kb),
```

Comment:
I should read the spec again to see if this can reach/exceed 1MB, since dmidecode logic would show MB again there.

Reply:
OK I see now, usually you don't write a math formula with mixed units and no space surrounding the operator.  "32 GB-1 MB"  means 32GB minus 1MB.  0x7FFFF is 32,767 in decimal.  The high bit is not set (the 0x8000 bit) and this signifies MB.  So we are talking about 32 GB (32,768) minus a MB which is 32,767 or greater meaning the greatest value possible is 0x7FFE or 32,766 decimal.

The original dmidecode code **normalizes** the MB value to KB with this line:
```cpp
		if (!(code & 0x8000))
			s.l <<= 10;
```

Note: << 10 multiplies by 1024 decimal.

Then the `dmi_print_memory_size()` function operates on the KB value and derives units.  The output will be either MB or GB (up to 32,766).